starknet-rpc-test/estimate_fee.rs

```diff
@@ -103,13 +105,30 @@ async fn works_ok(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error>
     let estimates =
         rpc.estimate_fee(&vec![invoke_transaction, invoke_transaction_2], BlockId::Tag(BlockTag::Latest)).await?;
 
-    // TODO: instead execute the tx and check that the actual fee are the same as the estimated ones
+    let tx = BroadcastedInvokeTransaction {
+        max_fee: FieldElement::from(210u16),
+        signature: vec![],
+        nonce: FieldElement::ZERO,
+        sender_address,
+        calldata,
+        is_query: false,
+    };
+
+    let invoke_transaction = BroadcastedTransaction::Invoke(tx.clone());
+
+    let invoke_transaction_2 =
+        BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction { nonce: FieldElement::ONE, ..tx });
+
+    let simulations = rpc
```

Comment:
can we execute it instead of doing the simulation? i think that would be a better test, wdyt?

Reply:
@apoorvsadana Thank you for your comment. Could you please clarify how to check the actual fee after execution? Additionally, does "executing it" refer to calling JsonRpcClient.add_invoke_transaction? Thanks

Reply:
You can fetch the receipt to get the actual_fee. Yes, you need to call add_invoke_transaction

Reply:
@apoorvsadana, Thank you for your help. do you know how long we should wait until `MaybePendingTransactionReceipt` gets the `Receipt` value? In my local test case, waiting for 20 seconds works fine. However, in the [rpc-tests](https://github.com/keep-starknet-strange/madara/actions/runs/8590075466/job/23537285842?pr=1526), it fails.